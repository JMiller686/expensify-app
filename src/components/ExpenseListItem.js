import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

//destructured props to use directly instead of off props object
const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    <div>
        <h3>{description}</h3>
        <p>{amount} - {createdAt}</p>
        <button onClick={() => {
            dispatch(removeExpense(id))
        }}>Remove</button>
    </div>
);


export default connect()(ExpenseListItem);



