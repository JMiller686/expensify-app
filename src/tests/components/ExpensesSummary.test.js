import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correcly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2300} />);
    expect(wrapper).toMatchSnapshot();
})

test('should correcly render ExpensesSummary with mulitple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={234530} />);
    expect(wrapper).toMatchSnapshot();
})