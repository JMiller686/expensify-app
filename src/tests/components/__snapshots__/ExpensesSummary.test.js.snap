// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correcly render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $23.00
  </p>
</div>
`;

exports[`should correcly render ExpensesSummary with mulitple expenses 1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     totalling 
    $2,345.30
  </p>
</div>
`;
